tests/contrib/io/matplotlib/test_simple_io.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited (“QuantumBlack”) name and logo
+# (either separately or in combination, “QuantumBlack Trademarks”) are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import matplotlib
+import numpy as np
+import pytest
+
+from kedro.contrib.io.matplotlib import MatplotlibWriter
+from kedro.io import DataSetError
+
+matplotlib.use('TkAgg')  # used to facilitate simple inclusion into kedro CI/CD
+
+
+def test_should_write_to_image_in_single_mode(tmpdir):
```

Comment:
I don't see the benefit in consolidating these tests in a class and think that it would probably be a deviation to how the other tests are written within the package. 

The name is deliberately specific as I believe tests should serve as a sort of secondary living documentation. If users were interested in how this class works, they could come in, look at this tests and easily how a matplotlib image is written in single mode.

This is based a little on the idea of specifications in behaviour driven development and the reasoning behind them too.